src/cpp/src/speculative_decoding/continuous_batching_for_speculative_decoding_impl.hpp

```diff
@@ -40,5 +40,50 @@ class ContinuousBatchingPipeline::ContinuousBatchingForSpeculativeDecodingImpl :
 protected:
     void finish_request(SequenceGroup::Ptr request);
     void _pull_awaiting_requests() override {};
+    bool eagle_mode_enabled = false;
+};
+
+class ContinuousBatchingPipeline::ContinuousBatchingForEagle3DecodingImpl
+    : public ContinuousBatchingPipeline::ContinuousBatchingForSpeculativeDecodingImpl {
+public:
+    ContinuousBatchingForEagle3DecodingImpl() = default;
+
+    ContinuousBatchingForEagle3DecodingImpl(const std::shared_ptr<ov::Model>& model,
+                                                 const Tokenizer& tokenizer,
+                                                 const GenerationConfig& generation_config,
+                                                 const SchedulerConfig& scheduler_config,
+                                                 const std::string& device,
+                                                 const ov::AnyMap& plugin_config,
+                                                 bool is_validation_mode_enabled) : ContinuousBatchingForSpeculativeDecodingImpl(
+                                                     model, tokenizer, generation_config,
+                                                     scheduler_config, device, plugin_config,
+                                                     is_validation_mode_enabled) {
+                                                        eagle_mode_enabled = true;
+                                                     };
```

Comment:
```suggestion
                                                     is_validation_mode_enabled), eagle_mode_enabled(true) {};
```
please also do something with code style. Now I see extra indentation.